src/prog_models/data_models/dmd.py

```diff
@@ -0,0 +1,452 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+from numbers import Number
+import numpy as np
+from scipy.interpolate import interp1d
+import types
+from warnings import warn
+
+from ..exceptions import ProgModelInputException
+from ..sim_result import SimResult, LazySimResult
+from .. import LinearModel, PrognosticsModel
+from . import DataModel
+
+
+class DMDModel(LinearModel, DataModel):
+    """
+    A subclass of LinearModel and DataModel that uses Dynamic Mode Decomposition to simulate a system throughout time.
+    
+    Given an initial state of the system (including internal states, outputs, and event_states), and the expected inputs throuhgout time, this class defines a model that can approximate the internal states, outputs, and event_states throughout time until threshold is met.
```

Comment:
I'm thinking we should edit this, as it's more appropriate for the original DMD, but doesn't make as much sense now that it's a more general data-driven model. 

Suggestion: "Given an initial state of the system and the expected inputs throughout time, this class defines a model that can approximate the dynamics of the system throughout time until threshold is met. This model can be fully data-driven (using from_data) or physics-enhanced (using from_model) where internal states of a high-fidelity model augment the purely data-driven method. "